pkg/stanza/operator/parser/container/parser.go

```diff
@@ -0,0 +1,359 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package container // import "github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/operator/parser/container"
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"regexp"
+	"sync"
+
+	jsoniter "github.com/json-iterator/go"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/entry"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/operator"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/operator/helper"
+)
+
+const dockerFormat = "docker"
+const crioFormat = "crio"
+const containerdFormat = "containerd"
+const recombineInternalID = "recombine_container_internal"
+const dockerPattern = "^\\{"
+const crioPattern = "^(?P<time>[^ Z]+) (?P<stream>stdout|stderr) (?P<logtag>[^ ]*) ?(?P<log>.*)$"
+const containerdPattern = "^(?P<time>[^ ^Z]+Z) (?P<stream>stdout|stderr) (?P<logtag>[^ ]*) ?(?P<log>.*)$"
+const logpathPattern = "^.*\\/(?P<namespace>[^_]+)_(?P<pod_name>[^_]+)_(?P<uid>[a-f0-9\\-]+)\\/(?P<container_name>[^\\._]+)\\/(?P<restart_count>\\d+)\\.log$"
+const logPathField = "log.file.path"
+
+var (
+	dockerMatcher     = regexp.MustCompile(dockerPattern)
+	crioMatcher       = regexp.MustCompile(crioPattern)
+	containerdMatcher = regexp.MustCompile(containerdPattern)
+	pathMatcher       = regexp.MustCompile(logpathPattern)
+)
+
+var (
+	logFieldsMapping = map[string]string{
+		"stream": "attributes.log.iostream",
+	}
+	k8sMetadataMapping = map[string]string{
+		"container_name": "attributes.k8s.container.name",
+		"namespace":      "attributes.k8s.namespace.name",
+		"pod_name":       "attributes.k8s.pod.name",
+		"restart_count":  "attributes.k8s.container.restart_count",
+		"uid":            "attributes.k8s.pod.uid",
+	}
+)
+
+// Parser is an operator that parses Container logs.
+type Parser struct {
+	helper.ParserOperator
+	recombineParser         operator.Operator
+	format                  string
+	json                    jsoniter.API
+	addMetadataFromFilepath bool
+	crioLogEmitter          *helper.LogEmitter
+	asyncConsumerStarted    bool
+	crioConsumers           *sync.WaitGroup
+	parserLock              sync.Mutex
+}
+
+// Process will parse an entry of Container logs
+func (p *Parser) Process(ctx context.Context, entry *entry.Entry) error {
+
+	var err error
+	format := p.format
+	if format == "" {
+		format, err = p.detectFormat(entry)
+		if err != nil {
+			return fmt.Errorf("failed to detect a valid container log format: %w", err)
+		}
+	}
+
+	switch format {
+	case dockerFormat:
+		return p.ParserOperator.ProcessWithCallback(ctx, entry, p.parseDocker, p.handleAttributeMappings)
+	case containerdFormat:
+		return p.ParserOperator.ProcessWithCallback(ctx, entry, p.parseContainerd, p.handleAttributeMappings)
+	case crioFormat:
+		if !p.asyncConsumerStarted {
```

Comment:
Since `Process` could be called multiple times in parallel, I think this could panic. 

Either way, a `sync.Once` might be simpler and/or more efficient.